lms/services/lti_launch.py

```diff
@@ -0,0 +1,102 @@
+from lms.models import ApplicationInstance, LTIParams, LTIUser, User
+from lms.services.application_instance import ApplicationInstanceService
+from lms.services.assignment import AssignmentService
+from lms.services.grading_info import GradingInfoService
+from lms.services.grouping import GroupingService
+from lms.services.lti_role_service import LTIRoleService
+
+
+class LTILaunchService:
+    # pylint:disable=too-many-instance-attributes,too-many-arguments
+    def __init__(
```

Comment:
This list is enormous. 

The services/dependencies are the same as in the view but here they are listed explicitly.

We could try to get "parent"  things from the "children"· 

Assignment->Course->Application Instance.


I reckon that will be easier to refactor with this service in place.